addons/web_editor/controllers/main.py

```diff
@@ -519,8 +520,9 @@ def shape(self, module, filename, **kwargs):
         """
         svg = None
         if module == 'illustration':
-            attachment = request.env['ir.attachment'].sudo().search([('url', '=like', request.httprequest.path), ('public', '=', True)], limit=1)
-            if not attachment:
+            path_end = request.httprequest.path.split('/')[-1]  # route ensures there are slashes
+            attachment = request.env['ir.attachment'].sudo().browse(unslug(path_end)[1])
```

Comment:
The end of the path is already parsed into the `filename` parameter by url-matching defined in the route so no need to get it manually ;)